core/src/wallet/ripple/api_impl/RippleLikeTransactionApi.cpp

```diff
@@ -43,6 +43,38 @@
 
 namespace ledger {
     namespace core {
+        // a helper to write VLE fields
+        bool writeLengthPrefix(BytesWriter& writer, uint64_t size) {
+            // encoding here: https://developers.ripple.com/serialization.html#length-prefixing
+            if (size <= 192) {
+                writer.writeByte(size & 0xFF);
```

Comment:
In theory yes. In practice, we must be careful.

According to [this], about standard conversions:

> Otherwise, if the conversion is between numeric types of the same kind (integer-to-integer or floating-to-floating), the conversion is valid, but the value is implementation-specific (and may not be portable).

In my case, it was there mostly to exhibit what part of `size` we’re interested in. We can clearly see here that we are passing 8-bit thanks to the `0xFF` constant.

You know how I feel about that kind of coercions… If you really think we should get rid of that and, I can remove it for sure.

Reply:
I was saying that to simplify the code :trollface: 
I understand, you want emphasize on passing only 8-bit ... alright then

Reply:
I think I could compromise to go into your direction would be to `static_cast<uint8_t>`. It would then prevent a useless mask (if the compiler is smart enough it will remove it… but we don’t actually know) and the `static_cast` should be enough to me to emphasize on the 8-bit.